src/raster_programs.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{errors::*, utils::_last_null_pointer_err, Dataset};
+use gdal_sys::GDALBuildVRTOptions;
+use libc::{c_char, c_int};
+use std::{
+    borrow::Borrow,
+    ffi::CString,
+    path::Path,
+    ptr::{null, null_mut},
+};
+
+/// Wraps a [GDALBuildVRTOptions] object.
+///
+/// [GDALBuildVRTOptions]: https://gdal.org/api/gdal_utils.html#_CPPv419GDALBuildVRTOptions
+struct BuildVRTOptions {
+    c_options: *mut GDALBuildVRTOptions,
+}
+
+impl BuildVRTOptions {
+    /// See [GDALBuildVRTOptionsNew].
+    ///
+    /// [GDALBuildVRTOptionsNew]: https://gdal.org/api/gdal_utils.html#_CPPv422GDALBuildVRTOptionsNewPPcP28GDALBuildVRTOptionsForBinary
+    fn new(args: Vec<String>) -> Result<Self> {
+        // Convert args to CStrings to add terminating null bytes
+        let mut cstr_args = Vec::<CString>::new();
```

Comment:
Untested, but:

```rust
let cstr_args = args.into_iter()
    .map(|arg| CString::new(arg))
    .collect::<Result<Vec<_>>>()?;
```

This looks nicer and (as a minor point) pre-allocates the destination `Vec`.